### PR TITLE
Document to describe networking requirements for vendors replacing openshift-sdn

### PR DESCRIPTION
**Pull request by rajatchopra**
@danmcp 
@mcurry-rh

### CODE REVIEW

File: docs/openshift_networking_requirements.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# OpenShift networking requirements
+Guidelines for a third party network plugin for OpenShift
```
Comment (pweil-) [inline docs/openshift_networking_requirements.md]: missing period

File: docs/openshift_networking_requirements.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Basic requirements
+OpenShift networking has certain requirements over and above kubernetes essentials. The basic kubernetes requirements can be found [here](https://github.com/kubernetes/kubernetes/blob/release-1.3/docs/design/networking.md).
+
+## CNI is the recommended way
+
+Any external networking solution can be used to plumb networking for openshift as long as it follows the 'CNI' spec. Then, openshift needs to be launched with 'networkPluginName: "cni"' in the master/node config yaml files.
```
Comment (pweil-) [inline docs/openshift_networking_requirements.md]: example master config snippet would be nice

File: docs/openshift_networking_requirements.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+Guidelines for a third party network plugin for OpenShift
+
+## Basic requirements
+OpenShift networking has certain requirements over and above kubernetes essentials. The basic kubernetes requirements can be found [here](https://github.com/kubernetes/kubernetes/blob/release-1.3/docs/design/networking.md).
+
+## CNI is the recommended way
+
+Any external networking solution can be used to plumb networking for openshift as long as it follows the 'CNI' spec. Then, openshift needs to be launched with 'networkPluginName: "cni"' in the master/node config yaml files. 
+
+When done through ansible, provide sdn_network_plugin_name=cni as the option while installing openshift. Be aware that openshift ansible installation allows a firewall passthrough for the VxLAN port (4789), so if a plugin needs other ports (for management/control/data) to be open, then the installer needs to be changed suitably.
```
Comment (pweil-) [inline docs/openshift_networking_requirements.md]: should mention that this goes in the inventory file.  Does it need to be in a specific group?  Example might also be helpful here.

File: docs/openshift_networking_requirements.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+   - Each namespace should be treated like a tenant where its pods and services are isolated from another project's pods/services
+   - *Support exists for operations like merge/join networks even when they belong to different namespaces
+
+2. Certain services in the cluster will be run as infrastructure services. e.g. Load balancer, registry, DNS server(skydns). The plugin should allow for a 'global' tenant which is-accessible-by/can-access all pods of the cluster. For example, a load balancer can run in two modes - private and global. The global load balancer should have access to all tenants/namespaces of the cluster. A private load balancer is one that is launched as a pod by a particular namespace, and this should obey tenant isolation rules.
+
+3. *Access to all pods from the host - particularly important if kube-proxy is used by the SDN solution to support kubernetes services. Please note that iptables based kube-proxy will be enabled by default in openshift. This will have to be overridden specially if the plugin wants a different behaviour.
```
Comment (pweil-) [inline docs/openshift_networking_requirements.md]: how to override?

**Comment by rajatchopra** [comment]
@pweil- Fixed according to the feedback. Thanks for the review.

File: docs/openshift_networking_requirements.md (hunk 2 above)
Comment (rajatchopra) [inline docs/openshift_networking_requirements.md]: Example has been added below

File: docs/openshift_networking_requirements.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# OpenShift networking requirements
+This document presents the guidelines for a third party network plugin to work with OpenShift.
+
+## Basic requirements
+OpenShift networking has certain requirements over and above kubernetes essentials. The basic kubernetes requirements can be found [here](https://github.com/kubernetes/kubernetes/blob/release-1.3/docs/design/networking.md).
+
+## CNI is the recommended way
+
+Any external networking solution can be used to plumb networking for openshift as long as it follows the 'CNI' spec. Then, openshift needs to be launched with 'networkPluginName: "cni"' in the master/node config yaml files.
```
Comment (danmcp) [inline docs/openshift_networking_requirements.md]: Should use OpenShift consistently (instead of openshift).

**Pull request merged**